src/components/input/input-animations.spec.js

```diff
@@ -1,70 +1,73 @@
 describe('md-input-container animations', function() {
-  var $rootScope, $compile, $animateCss, $material, $$mdInput,
-    el, pageScope, invalidAnimation, messagesAnimation, messageAnimation,
-    cssTransitionsDisabled = false, lastAnimateCall;
+  var $rootScope, $compile, $material, $$mdInput, $window, $animate, $rootElement, $document, $timeout,
+    el, root, body, pageScope, computedStyle;
 
   // Load our modules
   beforeEach(module('ngAnimate', 'ngMessages', 'material.components.input', 'material.components.checkbox'));
 
   // Run pre-test setup
-  beforeEach(decorateAnimateCss);
   beforeEach(injectGlobals);
   beforeEach(setupVariables);
 
   // Run after-test teardown
   afterEach(teardown);
 
-  it('set the proper styles when showing messages on an input', function() {
+  it('set the proper styles when showing messages on an input', performInputAnimationTests);
+  it('set the proper styles when showing messages on an input with animations disabled', function() {
+    $animate.enabled(false);
+    performInputAnimationTests();
+    $animate.enabled(true);
+  });
+
+  function performInputAnimationTests() {
     compile(
       '<form name="testForm">' +
       '  <md-input-container>' +
       '    <input name="foo" ng-model="foo" required ng-pattern="/^1234$/" />' +
       '    <div class="errors" ng-messages="testForm.foo.$error">' +
-      '      <div ng-message="required">required</div>' +
-      '      <div ng-message="pattern">pattern</div>' +
+      '      <div ng-message="required" style="transition: 0s none">required</div>' +
+      '      <div ng-message="pattern" style="transition: 0s none">pattern</div>' +
       '    </div>' +
       '  </md-input-container>' +
       '</form>'
     );
 
     var container = el.find('md-input-container'),
       input = el.find('input'),
-      doneSpy = jasmine.createSpy('done');
+      errors;
+      
 
     // Mimic the real validations/animations that fire
 
     /*
      * 1. Set to an invalid pattern but don't blur (so it's not invalid yet)
      *
-     * Expect nothing to happen ($animateCss called with no options)
+     * Expect nothing to happen (message is hidden)
      */
 
     setFoo('asdf');
-    messageAnimation.enter(getError(), doneSpy);
     flush();
-
-    expectError(getError(), 'pattern');
-    expect(doneSpy).toHaveBeenCalled();
+    errors = getError();
+    expectError(errors, 'pattern');
     expect(container).not.toHaveClass('md-input-invalid');
-    expect(lastAnimateCall).toEqual({element: getError(), options: {}});
+    computedStyle = $window.getComputedStyle(errors[0]);
```

Comment:
The previous tests weren't actually testing the component's animation procedure. 

For example:

`messageAnimation.leave(patternError, doneSpy);` was called and then `  expect(lastAnimateCall.options.event).toEqual('leave');`. It's a assertion, but didn't actually test whether the component was doing it automatically.

Actually testing the element `opacity` and `margin values` at each state of interaction ensures the object appears as it should.